libs/core/src/lib/fixed-card-layout/fixed-card-layout.component.ts

```diff
@@ -63,6 +75,92 @@ class CardDropped {
 
 type CardColumn = CardDefinitionDirective[];
 
+@Directive({ selector: '[fdGroup]' })
+export class DndGroup {
+    /** @hidden */
+    _onDndItemFocus$ = new Subject<[number, number]>();
+
+    constructor(public cdr: ChangeDetectorRef) {}
+
+    /** add focus after moving elements with keyboard */
+    focusDndItem(groupIndex: number, itemIndex: number): void {
+        this.cdr.detectChanges();
+        this._onDndItemFocus$.next([groupIndex, itemIndex]);
+    }
+}
+
+@Directive({ selector: '[fdItem]' })
+export class DndItem implements OnInit {
+    /** item index in group(column) */
+    @Input()
+    itemIndex: number;
+  
+    /** group(column) index */
+    @Input()
+    groupIndex: number;
+
+    constructor(
+        private dndGroup: DndGroup, 
+        private elementRef: ElementRef, 
+        private cardLayout: FixedCardLayoutComponent
+    ) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.dndGroup._onDndItemFocus$.subscribe(([groupIndex, itemIndex]) => {
+            if (this.groupIndex === groupIndex && this.itemIndex === itemIndex) {
+              this.elementRef.nativeElement.focus();
+            }
+        });
+    }
+    
+    /** disabled possibility to move card */
+    @HostListener('keyup', ['$event'])
+    onKeyUp(event: KeyboardEvent): void {
+        if (KeyUtil.isKeyCode(event, CONTROL)) {
+            this.cardLayout._enableKeyboard = false;
+            this.dndGroup._onDndItemFocus$.unsubscribe();
+            this.dndGroup._onDndItemFocus$ = new Subject<[number, number]>();
+        }
+    }
+
+    /** allow card movement using keyboard */
+    @HostListener('keydown', ['$event'])
+    onKeyDown(event: KeyboardEvent): void {
+        const group = this.cardLayout.columns[this.groupIndex];
+        if (KeyUtil.isKeyCode(event, CONTROL)) {
+            this.cardLayout._enableKeyboard = true
+        }
+        if (KeyUtil.isKeyCode(event, RIGHT_ARROW) 
+        && this.cardLayout._enableKeyboard 
+        && this.cardLayout.columns.length !== this.groupIndex + 1) {
+            event.preventDefault();
+            const nextGroup = this.cardLayout.columns[this.groupIndex + 1];
+            const nextGroupIndex = this.groupIndex + 1;
+            transferArrayItem(group, nextGroup, this.itemIndex, 0);   
+            this.dndGroup.focusDndItem(nextGroupIndex, 0);
+        } else if (KeyUtil.isKeyCode(event, DOWN_ARROW) 
```

Comment:
So much repeating block `if (...) else if (...)` look weirdo. 

Could we make it like 

```
if (key up) { ... }

if (key down) { ... }

if (key left) { ... }
```

Reply:
done